### PR TITLE
Revamp Announcement Dialog 

### PR DESCRIPTION
**Pull request by neuodev**
https://emurgo.atlassian.net/browse/YOEXT-667

## Behavior update 
- For new users the `revamp` will be the default theme right from the beginning.
- For existing users, will see the  announcement badge and when they click it they will be prompted to switch to the revamp 
- The behavior only works with `Dev` and `Nightly` builds. **Should be disabled with the revamp release** 
## Preview
<details>
 <summary>Screenshots</summary>
Screen Shot 2022-05-31 at 11 51 16 AM
</details>

### CODE REVIEW

**Comment by neuodev** [comment]
@yushih  Please don't merge it. Just review it

**Review by yushih — APPROVED** [review]

**Comment by neuodev** [comment]
After merging #3000  into this branch turns out there are some fixes that need to be made

**Comment by yushih** [comment]
@AhmedIbrahim336 np and thanks. Please let me know when it's done.

**Comment by neuodev** [comment]
@yushih  this PR is ready for review

**Review by Nebyt — DISMISSED** [review]

File: packages/yoroi-extension/app/components/wallet/staking/dashboard/StakingDashboard.js
Comment (yushih) [inline packages/yoroi-extension/app/components/wallet/staking/dashboard/StakingDashboard.js]: Since you are only using the `currentTheme` to decide whether to show the revamp announcement, it is not necessary to pass in the theme but only a flag to indicate whether the current theme is revamp. This is the "Law of Demeter" of software design.

File: packages/yoroi-extension/app/containers/profile/LanguageSelectionPage.js
Comment (yushih) [inline packages/yoroi-extension/app/containers/profile/LanguageSelectionPage.js]: Why do it here instead of changing the default theme in `BaseProfileStore`?

File: packages/yoroi-extension/app/containers/wallet/Wallet.js
Comment (yushih) [inline packages/yoroi-extension/app/containers/wallet/Wallet.js:247]: Why are we changing the order of `{this.renderOverlay()}` to after `{this.props.children}`?

**Review by yushih — CHANGES_REQUESTED** [review]

File: packages/yoroi-extension/app/containers/wallet/Wallet.js
Comment (neuodev) [reply packages/yoroi-extension/app/containers/wallet/Wallet.js:247]: No specific reason. I think this diff was made while I was testing something.

File: packages/yoroi-extension/app/containers/profile/LanguageSelectionPage.js
Comment (neuodev) [reply packages/yoroi-extension/app/containers/profile/LanguageSelectionPage.js]: Nice! I did it

**Review by neuodev — COMMENTED** [review]
/check

**Comment by nistadev** [comment]
Is this PR still relevant? @neuodev @vsubhuman

**Comment by zuzunker** [comment]
Closed in favour of https://github.com/Emurgo/yoroi-frontend/pull/3282

**Pull request closed without merging**
